internal/pkg/cli/app_show.go

```diff
@@ -242,7 +255,22 @@ func (o *showAppOpts) retrieveProjects() ([]string, error) {
 	return projNames, nil
 }
 
-func (o *showAppOpts) retrieveApplications() ([]string, error) {
+func (o *showAppOpts) retrieveLocalApplication() ([]string, error) {
+	localApps, err := o.ws.Apps()
+	if err != nil {
+		return nil, err
+	}
+	if len(localApps) == 0 {
+		return nil, errors.New("no application found")
+	}
+	var localAppNames []string
+	for _, app := range localApps {
+		localAppNames = append(localAppNames, app.AppName())
+	}
```

Comment:
I normally pre-allocate the slice to avoid the possibility of multiple memory allocation when the number of items outgrow the capacity of the underlying array that backs the slice:

```
localAppNames := make([]string, 0, len(localApps))
// the same
```